### PR TITLE
Refactor to re-use CV-based image segmentation and normals

### PR DESCRIPTION
**Pull request by markus-hinsche**
**CR:** https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/1543

**Changes:**
- Prevent redundant `parse_depth()` calculation by maintaining an np.array `depthmap_arr`
- Improve Depthmap constructor interface: now support `create_from_file` (ZIP) and `create_from_array` (np.array)
- Introduce `Depthmap.create_from_array()`
- Introduce `EDA_RGBD_segmentationCV.ipynb` to analyze RGBD pickle files with depthmap toolkit
- Fix: toolkit fails to ignore hidden files like `.DS_Store`
- Introduce `is_google_tango_resolution()`
- Remove unused `convert_3d_to_2d`
- Use cgm-ml/data/export as default export directory instead of using code directories
- Making export more robust
- Making toolkit.py callable from cgm-ml dir: `python src/common/depthmap_toolkit/toolkit.py src/common/depthmap_toolkit/tests/huawei_p40pro src/common/depthmap_toolkit/tests/huawei_p40pro/camera_calibration.txt` (no need to `cd src/common/depthmap_toolkit` anymore)

**How Has This Been Tested?**
- existing unit tests still run
- running depthmap toolkit locally still works

**Checklist:**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests
- [x] New and existing unit tests pass locally with my changes
- [x] My Code is review by 1 People

### CODE REVIEW

**Comment by lvonasek** [comment]
I will check it detaily when I am back from vacation (Thursday). Please do not merge it, there are q few things which need to be changed.

**Comment by markus-hinsche** [comment]
I am currently working on speeding this up because the for loops over all pixels take quite long for me (4 to 20 seconds per depthmap for the more advanced operations that have to be done per pixel). 

@lvonasek You can already review the changes in this PR, but please don't make commits into this branch, as I have uncommitted/unfinished work which is based on this.

**Review by lvonasek — CHANGES_REQUESTED** [review]
Here is a change of the behavior, Z-axis has a different coordinate system in comparison with the previous version.
```
        res = [-res[0], res[1], res[2]]
        if is_google_tango_resolution(self.width, self.height):
            res = [-res[0], -res[1], -res[2]]
```

Do not use try and catch for index error, fix the if statements instead
```
        if tx - 1 < 0 or ty - 1 < 0:
            return 0.
        if tx + 1 >= self.width or ty + 1 >= self.height:
            return 0.

        # Get all neighbor depths
        try:
            depth_center = self.depthmap_arr[tx, ty]
            depth_x_minus = self.depthmap_arr[tx - 1, ty]
            depth_x_plus = self.depthmap_arr[tx + 1, ty]
            depth_y_minus = self.depthmap_arr[tx, ty - 1]
            depth_y_plus = self.depthmap_arr[tx, ty + 1]
        except IndexError:
            return 0.
```

File: src/common/depthmap_toolkit/depthmap.py [hunk 1]
```diff
@@ -161,20 +197,21 @@ def convert_2d_to_3d_oriented(self, sensor: int, x: float, y: float, depth: floa
...
             return res
 
-        # special case for Google Tango devices with different rotation
-        if self.width == 180 and self.height == 135:
-            res = [res[0], -res[1], res[2]]
-        else:
-            res = [-res[0], res[1], res[2]]
+        res = [-res[0], res[1], res[2]]
+        if is_google_tango_resolution(self.width, self.height):
+            res = [-res[0], -res[1], -res[2]]
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/depthmap.py:202]:
```suggestion
            res = [-res[0], -res[1], res[2]]
```

**Review by lvonasek — APPROVED** [review]
LGTM

**Pull request merged**